cache/lru_cache.cc

```diff
@@ -241,17 +258,36 @@ void LRUCacheShard::LRU_Insert(LRUHandle* e) {
     e->prev->next = e;
     e->next->prev = e;
     e->SetInHighPriPool(true);
+    e->SetInLowPriPool(false);
     high_pri_pool_usage_ += e->total_charge;
     MaintainPoolSize();
-  } else {
+  } else if (low_pri_pool_ratio_ > 0 && (e->IsLowPri() || e->HasHit())) {
     // Insert "e" to the head of low-pri pool. Note that when
     // high_pri_pool_ratio is 0, head of low-pri pool is also head of LRU list.
-    e->next = lru_low_pri_->next;
-    e->prev = lru_low_pri_;
+    if (lru_low_pri_ == &lru_) {
+      // The low-pri pool is empty.
+      e->next = lru_bottom_pri_->next;
+      e->prev = lru_bottom_pri_;
+    } else {
+      e->next = lru_low_pri_->next;
+      e->prev = lru_low_pri_;
+    }
```

Comment:
Similar question to what we discussed earlier: do we need this branching?

Reply:
if lru bottom pri pool is not empty but lru low pri pool is, we need this branching, right?